vector/bool.go

```diff
@@ -64,6 +64,16 @@ func Or(a, b *Bool) *Bool {
 	return NewBool(bits, nulls)
 }
 
+func Not(vec *Bool) *Bool {
+	bits := bitvec.Not(vec.Bits)
+	if vec.Nulls.IsZero() {
+		return NewBool(bits, bitvec.Zero)
+	}
+	// Flip true/null values to false/null.
+	bits = bitvec.And(bits, bitvec.Not(bitvec.And(bits, vec.Nulls)))
```

Comment:
Nit: No need for two logical conjunctions.
```
A & ^(A & B) = A & (^A | ^B)        By De Morgan's law.
             = (A & ^A) | (A & ^B)  By the distributive property.
             = A & ^B               Because A & ^A is always false.
```

```suggestion
	bits = bitvec.And(bits, bitvec.Not(vec.Nulls))
```